core/src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -183,6 +183,16 @@ private void extract(PluginHandle pluginHandle, Terminal terminal, Path pluginFi
             throw new IOException("plugin directory " + extractLocation.toAbsolutePath() + " already exists. To update the plugin, uninstall it first using 'remove " + pluginHandle.name + "' command");
         }
 
+        Path pluginConfigDirectory = pluginHandle.configDir(environment);
+        if (Files.exists(pluginConfigDirectory) && !Files.isDirectory(pluginConfigDirectory)) {
```

Comment:
not needed, one test is covering this already and returning an appropriate exception